packages/snjs/lib/services/settings_service/SNSettingsService.ts

```diff
@@ -0,0 +1,32 @@
+import { PureService } from '../pure_service';
+
+import { SNApiService } from '../api/api_service';
+import { SettingsProvider } from './SettingsProvider';
+import { SettingsGateway } from './SettingsGateway';
+import { SNSessionManager } from '../api/session_manager';
+
+export class SNSettingsService extends PureService {
+  private _provider!: SettingsProvider;
+
+  constructor(
+    private readonly sessionManager: SNSessionManager,
+    private readonly apiService: SNApiService
+  ) {
+    super();
+  }
+
+  initializeFromDisk() {
+    const getUser = () => this.sessionManager.getUser();
+    this._provider = new SettingsGateway(this.apiService, getUser);
```

Comment:
Feel like there could be a retention cycle here. Why not pass the user directly to the gateway? Or pass the session manager?

The cycle is that `this` retains provider, provider retains `this` via the `getUser` function.

Reply:
When you release your reference in deinit in this class, the settings service will likely not be deallocated because it's still being held onto by the gateway.

Reply:
Fixed.